docs/developer/contributing/development-ci-metrics.asciidoc

```diff
@@ -0,0 +1,65 @@
+[[ci-metrics]]
+== CI Metrics
+
+In addition to run our tests CI collects metrics about Kibana. These metrics are sent to an external service which catalogs and compares them to the metrics from other branches so that PR authors can better quantify the impact their changes have.
+
+
+[[ci-metric-types]]
+=== Metric types
+
+
+[[ci-metric-types-bundle-size-metrics]]
+==== Bundle size
+
+These metrics help contributors know how they are impacting the size of the bundles Kibana creates, and help make sure that Kibana loads as fast as possible.
+
+[[ci-metric-page-load-bundle-size]] `page load bundle size` ::
+The size of the entry file produced for each bundle/plugin. This file is always loaded on every page load, so it should be as small as possible. To reduce this metric you can put any code that isn't necessary on every page load behind an https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#Dynamic_Imports[`async import()`].
```

Comment:
Should we link to https://github.com/elastic/kibana/blob/master/src/core/MIGRATION.md#keep-kibana-fast, or move that particular part to docs? cc @joshdover

Reply:
I like the idea of linking there

Reply:
Oh, didn't realize those weren't already in the docs. We should probably move that to the docs and link to there then. Linking to a section of the MIGRATION.md file feels like it's going to break pretty quickly.

Reply:
I don't think we should block this PR on that change, but we should add the link once those docs are moved.